go.mod

```diff
@@ -2,7 +2,7 @@ module github.com/operator-framework/operator-marketplace
 
 go 1.23.0
 
-toolchain go1.23.4
+toolchain go1.24.1
```

Comment:
We probably need to do this one manually due this change 

/hold

Reply:
toolchain go1.24 should be ok, no? even with 1.23?

Reply:
I guess, if it passes CI it should be ok?

Reply:
That is true, I was concern because usually we cannot update go version 
maybe that is good re-check with @tmshort 

This one seems fine we change the go-version

Reply:
I think this is actually an unintended consequence of dependabot using the latest version of tooling for this operation.  
When I bump this locally, I get no golang or toolchain version bump. 
Maybe we just do this one manually?

Reply:
Now I'm really confused.  
ART builds https://github.com/operator-framework/operator-marketplace/blob/master/Dockerfile, and if I replicate this PR on my local and build with it (taking some steps to ensure that my GOTOOLCHAIN is set to local for the build) ... it works fine. 
I guess I expected it to try to download the toolchain and fail, and fail to build.  
🤷

Reply:
Reading [this](https://go.dev/doc/toolchain#switch) as well as the fact that my local `go get sigs.k8s.io/controller-runtime@v0.20.4` doesn't bump the toolchain, I interpret this toolchain bump as arbitrary and not needed by any dependency.  
Since it is arbitrary, the compile process ignores it, which means in our 1.23.6-constrained, local environment it will still compile.

Reply:
I think we could create a manual PR without the Go version bump.

However, regards it be passing in the CI, I either do not understand. I would actually expect it to fail as well until we be able to use up go version.